UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.CleanUpJob/Helpers/AzureFileSystemHelper.cs

```diff
@@ -0,0 +1,88 @@
+
+                foreach (var subFolderItem in subFolder)
+                {
+                    string subFolderName = new DirectoryInfo(subFolderItem).Name;
+                    bool isSubFolderDateFolder = DateTimeExtensions.IsValidDate(subFolderName, out subFolderDateTime);
+                    
+                    if (isSubFolderDateFolder)
+                    {
+                        DateTime subFolderDate = new DateTime(subFolderDateTime.Year, subFolderDateTime.Month, subFolderDateTime.Day);
+                        
+                        if (subFolderDate <= historicDate)
+                        {
+                            DirectoryInfo di = new DirectoryInfo(subFolderItem);
+
+                            foreach (DirectoryInfo subDirectory in di.GetDirectories())
+                            {
+                                var batchId = subDirectory.Name;
+                                string scsResponseFileName = new DirectoryInfo(batchId).Name + ".json";
+
+                                CloudBlockBlob cloudBlockBlob = azureBlobStorageClient.GetCloudBlockBlob(scsResponseFileName, storageAccountConnectionString, containerName);
+                                var response = await cloudBlockBlob.DeleteIfExistsAsync();
+
+                                if (response)
+                                {
+                                    subDirectory.Delete(true);
+                                    logger.LogInformation(EventIds.HistoricSCSResponseFileDeleted.ToEventId(), "SCS response json file {ScsResponseFileName} deleted successfully from the container.", scsResponseFileName);
+                                }
+                                else
+                                {
+                                    logger.LogError(EventIds.HistoricSCSResponseFileNotFound.ToEventId(), "SCS response json file {ScsResponseFileName} not found in the container.", scsResponseFileName);
+                                }
+                            }
+
+                            di.Delete(true);
+                            deteleFolderStatus = true;
+                            logger.LogInformation(EventIds.HistoricDateFolderDeleted.ToEventId(), "Historic folder deleted successfully for Date:{subFolderName}.", subFolderName);
+                        }
+                    }
+                }
+
+                if (!deteleFolderStatus)
+                {
+                    logger.LogError(EventIds.HistoricDateFolderNotFound.ToEventId(), "Historic folder not found for Date:{historicDate}.", historicDate);
+                }
+                return true;
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(EventIds.DeleteHistoricFoldersAndFilesException.ToEventId(), ex, "Exception while deleteing historic folders and files with error {Message}", ex.Message);
+                return false;
+            }
+        }
+    }
+}
```

Comment:
There appears to be quite a bit of cyclic complexity in this method. Would it be possible to break it down into a number of methods? Each method could be named to describe the work its is doing which might make it easier to follow the logic/flow.

Interested to know if this piece of code needs to be excluded from code coverage as it feels like it is doing quite a bit of work with a number of steps which could go wrong/could be tested?

Reply:
Resolved. As discussed we have break it down into a number of methods.
Yes for this piece of code its need physical file path when we are doing unit testing so for this reason we have excluded from code coverage.